makefiles/cflags.inc.mk

```diff
@@ -57,6 +57,26 @@ ifeq ($(LTO),1)
   LINKFLAGS += $(LTOFLAGS) -ffunction-sections -fdata-sections
 endif
 
+# Check if linker supports `-z noexecstack`. Handle BUILD_IN_DOCKER separately,
+# as this is run in the host environment rather than inside the container. We
+# just hardcode this in the BUILD_IN_DOCKER case for now.
+ifneq (0,$(BUILD_IN_DOCKER))
```

Comment:
```suggestion
ifeq (1, $(BUILD_IN_DOCKER))
```

is easier to read

Reply:
I think this is consistent with the other checks. I want to avoid that e.g. `BUILD_IN_DOCKER=y` partially works.

Reply:
I just checked: In fact, the other instances did check like you suggested, so this would introduce inconsistent behavior.